### PR TITLE
Add default location setting (merge from JiscRDSS fork)

### PR DESCRIPTION
**Pull request by lower29**
This merges changes from https://github.com/JiscRDSS/archivematica-storage-service/pull/8 and https://github.com/JiscRDSS/archivematica-storage-service/pull/11 into the Artefactual fork for issue #211.

Code is unchanged from @sevein's original commits - the only thing I've done is combine the two PRs and fix a minor typo in the location template and a flake8 whitespace error in `storage_service/locations/models/location.py`.

### CODE REVIEW

**Comment by helenst** [comment]
I've rebased this against the latest qa/1.x

File: storage_service/locations/api/resources.py [hunk 1]
```diff
@@ -272,6 +276,26 @@ def get_objects(self, space, path):
         message = _('This method should be accessed via a versioned subclass')
         raise NotImplementedError(message)
 
+    def default(self, request, **kwargs):
```
Comment (jrwdunham) [inline storage_service/locations/api/resources.py:279]: Is there a reason that you can not using the `_custom_endpoint` decorator? If not, then maybe you should for module-internal consistency. If yes, could you add a brief comment indicating why?

File: storage_service/locations/api/resources.py [hunk 2]
```diff
@@ -465,6 +491,34 @@ def dehydrate_misc_attributes(self, bundle):
         # Serialize JSONField as dict, not as repr of a dict
         return bundle.obj.misc_attributes
 
+    def hydrate_current_location(self, bundle):
```
Comment (jrwdunham) [inline storage_service/locations/api/resources.py:498]: I think this should have a docstring to reduce the amount of tastypie knowledge that is required to understand this functionality. Also, I think the `LOGGER.debug` calls should be to `.info` and `.warning` in production, no?

File: storage_service/locations/models/location.py [hunk 3]
```diff
@@ -81,6 +83,12 @@ class Meta:
     objects = models.Manager()
     active = Enabled()
 
+    # Whether this location is the default location which is a global
+    # application setting stored in administration.models.Settings. See
+    # signal receivers unset_default_locaiton and set_default_location for
```
Comment (jrwdunham) [inline storage_service/locations/models/location.py:88]: typo

File: storage_service/locations/forms.py [hunk 4]
```diff
@@ -163,9 +163,11 @@ class Meta:
 
 
 class LocationForm(forms.ModelForm):
+    default = forms.BooleanField(required=False)
```
Comment (jrwdunham) [inline storage_service/locations/forms.py]: Can this have a help string, e.g., like the "Enabled" attribute has "True if space can be accessed"? 

Speaking of which, it seems confusing that it's not "True if location can be accessed" at https://github.com/artefactual/archivematica-storage-service/blob/qa/0.x/storage_service/locations/models/location.py#L75, no?

**Review by jrwdunham — CHANGES_REQUESTED** [review]
I've played around with this and it works as I expect. I have a couple of change requests related to user and developer documentation. But in terms of functionality it looks good to me.

File: storage_service/locations/forms.py (hunk 4 above)
Comment (sevein) [reply storage_service/locations/forms.py]: True but that'd need a new migration, right?

**Comment by sevein** [comment]
Ok, I've made the changes suggested!

**Review by helenst — COMMENTED** [review]
I'm happy with this, nothing to add beyond what @jrwdunham has already pointed out.

File: storage_service/locations/forms.py [hunk 5]
```diff
@@ -163,9 +163,11 @@ class Meta:
 
 
 class LocationForm(forms.ModelForm):
+    default = forms.BooleanField(required=False, help=_l("Set as global default location for its purpose"))
```
Comment (helenst) [inline storage_service/locations/forms.py]: This should be `label` rather than `help`

**Review by helenst — CHANGES_REQUESTED** [review]

**Comment by sevein** [comment]
I'll take a look when I'm back on Tuesday, unless you want to do it.

**Comment by helenst** [comment]
@sevein no worries, I will sort it out! :)

**Review by helenst — APPROVED** [review]

**Pull request merged**